test/meta_testing_tools.jl

```diff
@@ -0,0 +1,75 @@
+# This is tools for testing ChainRulesTestUtils itself
+# if they were less nasty in implementation we might consider moving them to a package
+# MetaTesting.jl
```

Comment:
yeah, i see why this is helpful to us in CRTestUtils, but i think down-the-road we want to move this to it's own package, since it's not CRTestUtils specific :)

Reply:
https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/62